internal/core/src/exec/expression/Expr.h

```diff
@@ -826,13 +828,33 @@ class SegmentExpr : public Expr {
             if (cached_index_chunk_id_ != i) {
                 Index* index_ptr = nullptr;
                 PinWrapper<const Index*> pw;
+                // Executor for JsonFlatIndex. Must outlive index_ptr. Only used for JSON type.
+                std::shared_ptr<
+                    index::JsonFlatIndexQueryExecutor<IndexInnerType>>
+                    executor;
 
                 if (field_type_ == DataType::JSON) {
                     auto pointer = milvus::Json::pointer(nested_path_);
 
-                    pw = segment_->chunk_scalar_index<IndexInnerType>(
-                        field_id_, pointer, i);
-                    index_ptr = const_cast<Index*>(pw.get());
+                    auto index =
```

Comment:
assign to existing pw variable instead.

Reply:
> assign to existing pw variable instead.

pw's type is not same as return type of chunk_scalar_index.